plugins/serializers/msgpack/msgpack_test.go

```diff
@@ -0,0 +1,237 @@
+package msgpack
+
+import (
+	"testing"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/metric"
+	"github.com/stretchr/testify/assert"
+)
+
+func isIdentical(t *testing.T, m1 telegraf.Metric, m2 *Metric) bool {
+	// Max precision in msgpack is nanoseconds
+	// https://github.com/msgpack/msgpack/blob/master/spec.md
+	if m1.Time().Truncate(time.Nanosecond) != m2.Time.time {
+		t.Logf("expected: %v, actual: %v", m1.Time().Truncate(time.Nanosecond), m2.Time.time)
+		return false
+	}
+
+	if m1.Name() != m2.Name {
+		t.Logf("expected: %v, actual: %v", m1.Name(), m2.Name)
+		return false
+	}
+
+	if len(m1.Tags()) != len(m2.Tags) {
+		t.Logf("expected: %v, actual: %v", m1.Tags(), m2.Tags)
+		return false
+	}
+
+	for k, v := range m1.Tags() {
+		if m2.Tags[k] != v {
+			t.Logf("expected: %v, actual: %v", m1.Tags(), m2.Tags)
+			return false
+		}
+	}
+
+	if len(m1.Fields()) != len(m2.Fields) {
+		t.Logf("expected: %v, actual: %v", m1.Fields(), m2.Fields)
+		return false
+	}
+
+	for k, v := range m1.Fields() {
+		if m2.Fields[k] != v {
+			t.Logf("expected: %v, actual: %v", m1.Fields(), m2.Fields)
+			return false
+		}
+	}
+
+	return true
+}
+
+func TestSerializeMetricFloat(t *testing.T) {
+	now := time.Now()
+	tags := map[string]string{
+		"cpu": "cpu0",
+	}
+	fields := map[string]interface{}{
+		"usage_idle": float64(91.5),
+	}
+	m, err := metric.New("cpu", tags, fields, now)
+	assert.NoError(t, err)
+
+	s := Serializer{}
+	var buf []byte
+	buf, err = s.Serialize(m)
+	assert.NoError(t, err)
+
+	m2 := &Metric{}
+	left, err := m2.UnmarshalMsg(buf)
+	assert.NoError(t, err)
+
+	assert.Equal(t, len(left), 0)
+
+	assert.Equal(t, true, isIdentical(t, m, m2))
```

Comment:
You can use `testutils.RequireMetricEqual()` or `RequireMetricsEqual()` in case you want to compare two or more metrics.

Reply:
Ack.

m2 here is not a telegraf.Metric but writing a simple converter is cleaner. I will change. And I also found the testutil.TestMetric function. I will update the code with it.